proc-macro/src/lib.rs

```diff
@@ -690,6 +697,25 @@ impl Literal {
         Literal::_new(escaped)
     }
 
+    pub fn c_string(cstr: &CStr) -> Self {
+        let mut escaped = "c\"".to_string();
+        for b in cstr.to_bytes() {
+            match *b {
+                b'\t' => escaped.push_str(r"\t"),
+                b'\n' => escaped.push_str(r"\n"),
+                b'\r' => escaped.push_str(r"\r"),
+                b'"' => escaped.push_str("\\\""),
+                b'\\' => escaped.push_str("\\\\"),
+                b'\x20'..=b'\x7E' => escaped.push(*b as char),
+                _ => {
+                    let _ = write!(escaped, "\\x{:02X}", b);
```

Comment:
It looks like you copied this from the implementation of `byte_string` above. Unlike byte strings, C strings are allowed to contain multibyte Unicode escapes so \x is not a good representation for any parts of the string that are valid UTF-8. (For example `b"\u{ffef}"` is not allowed but `c"\u{ffef}"` is.)

I will merge the PR as is to unbreak compilation, but would you be able to follow up with a second PR to match proc-macro2's implementation of `c_string` more closely?

Reply:
https://github.com/dtolnay/watt/pull/57